scripts/zones/Lower_Jeuno/npcs/Zalsuhm.lua

```diff
@@ -3,114 +3,110 @@
 --  NPC: Zalsuhm
 -- Standard Info NPC
 -----------------------------------
-require("scripts/globals/equipment");
-require("scripts/globals/quests");
+local ID = require("scripts/zones/Lower_Jeuno/IDs")
+require("scripts/globals/equipment")
+require("scripts/globals/quests")
+require("scripts/globals/status")
+-----------------------------------
 
 function getQuestId(mainJobId)
 
-    return (UNLOCKING_A_MYTH_WARRIOR - 1 + mainJobId);
+    return (UNLOCKING_A_MYTH_WARRIOR - 1 + mainJobId)
 
-end;
+end
 
 function onTrade(player,npc,trade)
-
-    -- printf("LowerJeuno_Zalsuhm.onTrade() - ");
-
     if (trade:getItemCount() == 1) then
         for i, wepId in pairs(BaseNyzulWeapons) do
-            if (trade:hasItemQty(wepId, 1)) then
+            if trade:hasItemQty(wepId, 1) then
```

Comment:
So there's an npcUtil function for this. npcUtil.tradeHas. It takes an optional "hasExactly" boolean so you can reduce all these checks down to 1.

Reply:
I am under the impression that @wrenffxi made sure there were both `tradeHas()` and `tradeHasExactly()`, so the scripter can use `tradeHasExactly(item)` without having to worry about magic boolean optional params on `tradeHas()`.

(The boolean in `tradeHas()` is probably because `tradeHasExactly()` is a shorthand function to `tradeHas()`)